statemachine/callbacks.py

```diff
@@ -46,8 +46,8 @@ def __str__(self):
     def __lt__(self, other):
         return self.meta.priority < other.meta.priority
 
-    def __call__(self, *args, **kwargs):
-        return self._callback(*args, **kwargs)
+    async def __call__(self, *args, **kwargs):
+        return await self._callback(*args, **kwargs)
```

Comment:
Since the library automatically calls this, I've chosen this approach to avoid internal conditionals for sync/async code. Internally, I've implemented everything using async.